src/librustdoc/html/format.rs

```diff
@@ -1085,12 +1085,21 @@ impl Function<'_> {
 }
 
 impl clean::Visibility {
-    crate fn print_with_space<'tcx>(self, tcx: TyCtxt<'tcx>) -> impl fmt::Display + 'tcx {
+    crate fn print_with_space<'tcx>(
+        self,
+        tcx: TyCtxt<'tcx>,
+        item_did: LocalDefId,
+    ) -> impl fmt::Display + 'tcx {
         use rustc_span::symbol::kw;
 
         display_fn(move |f| match self {
             clean::Public => f.write_str("pub "),
             clean::Inherited => Ok(()),
+            clean::Visibility::Restricted(did)
+                if did.index == tcx.parent_module_from_def_id(item_did).local_def_index =>
+            {
+                Ok(())
+            }
             clean::Visibility::Restricted(did) if did.index == CRATE_DEF_INDEX => {
```

Comment:
Can you also handle `pub(super)`?

nit: It might be worth making only a single match arm for `Restricted` and handling the special cases in there.

Reply:
Yes, I was planning on handling it once I get the basics sorted out (i.e. fixing the ICEs). And yeah it probably makes sense to collapse the arms.

Reply:
I now handle `pub(super)`.